src/main/java/io/mathdojo/useraccountservice/PostOrganisationsHandler.java

```diff
@@ -0,0 +1,55 @@
+package io.mathdojo.useraccountservice;
+
+import java.util.Optional;
+import java.util.logging.Level;
+
+import javax.validation.ConstraintViolationException;
+
+import com.microsoft.azure.functions.ExecutionContext;
+import com.microsoft.azure.functions.HttpMethod;
+import com.microsoft.azure.functions.HttpRequestMessage;
+import com.microsoft.azure.functions.HttpResponseMessage;
+import com.microsoft.azure.functions.HttpStatus;
+import com.microsoft.azure.functions.annotation.AuthorizationLevel;
+import com.microsoft.azure.functions.annotation.FunctionName;
+import com.microsoft.azure.functions.annotation.HttpTrigger;
+
+import io.mathdojo.useraccountservice.model.Organisation;
+import io.mathdojo.useraccountservice.model.requestobjects.AccountRequest;
+import io.mathdojo.useraccountservice.security.HTTPRequestSignatureVerificationEnabledHandler;
+import io.mathdojo.useraccountservice.services.OrganisationServiceException;
+
+public class PostOrganisationsHandler extends HTTPRequestSignatureVerificationEnabledHandler<AccountRequest, Organisation> {
+
+    @FunctionName("createOrganisation")
+    public HttpResponseMessage executePostForOrganisations(
+        @HttpTrigger(
+            name = "request", 
+            methods = { HttpMethod.POST }, 
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "organisations"
+        ) HttpRequestMessage<Optional<AccountRequest>> request,
+        ExecutionContext context) {
+
+            try {
+                Object handledRequest = handleRequest(request, request.getBody().get(), context);
+                if(handledRequest instanceof HttpResponseMessage) {
+                    return (HttpResponseMessage) handledRequest;
+                }
+                Organisation createdOrg = (Organisation) handledRequest;
```

Comment:
Is a Class cast exception possible here, and if so is it handled?

Reply:
I'm assuming you mean for line 39 only? If so, it is caught as part of the 500 error return branch:
https://github.com/math-dojo/user-account-service/blob/39ffd4dd68211600fa26405d69a7b430d88614e6/src/main/java/io/mathdojo/useraccountservice/PostOrganisationsHandler.java#L47